addons/website_slides_survey/controllers/survey.py

```diff
@@ -7,8 +7,8 @@
 class Survey(main.Survey):
     def _prepare_survey_finished_values(self, survey, answer, token=False):
         result = super(Survey, self)._prepare_survey_finished_values(survey, answer, token)
-        if answer.slide_id:
-            result['channel_id'] = answer.slide_id.channel_id
+        if answer.slide_partner_id and answer.slide_partner_id.slide_id:
```

Comment:
Is it really important to change this? Do we need the add the channel_id if you're no longer a member of this course?

Reply:
Indeed , this is used in the `Go Back to course` button , so when a user fails its last attempt it will be removed from the course and then this function is called so in the old behavior it would not return anything and thus the button doesn't show in the last attempt.

Reply:
How did it work before we made the change that archive subscriptions instead of deleting them then? As the membership would not exist either at this point?

Reply:
As far as I know , before the membership doesn't exist because it was archived but the answers still exist and not changed that's why it's valid to write answer.slide_id

Reply:
Can we do that instead then, remove the membership relationship but not the slide_id, that way we have the same behavior?

Reply:
Okay I will change it then

Reply:
@awa-odoo  i think there is a problem with this is that removing the relationship between user_input and slide_partner_id will make it impossible to retrieve the the user_inputs for some slide_partner